binding.gyp

```diff
@@ -48,7 +48,7 @@
           'xcode_settings': {
             'GCC_ENABLE_CPP_EXCEPTIONS': 'YES',
             'GCC_ENABLE_CPP_RTTI': 'YES',
-            'MACOSX_DEPLOYMENT_TARGET': '11.0',
+            'MACOSX_DEPLOYMENT_TARGET': '12.6',
```

Comment:
Why did you upgrade this version?

Reply:
In order to solve this warning:

```
ld: warning: object file (/Users/runner/work/pulsar-client-node/pulsar-client-node/pkg/mac/build-pulsar/install/lib/libpulsarwithdeps.a(io_win32.cc.o)) was built for newer macOS version (12.6) than being linked (11.0)
ld: warning: object file (/Users/runner/work/pulsar-client-node/pulsar-client-node/pkg/mac/build-pulsar/install/lib/libpulsarwithdeps.a(json_escaping.cc.o)) was built for newer macOS version (12.6) than being linked (11.0)
ld: warning: object file (/Users/runner/work/pulsar-client-node/pulsar-client-node/pkg/mac/build-pulsar/install/lib/libpulsarwithdeps.a(json_objectwriter.cc.o)) was built for newer macOS version (12.6) than being linked (11.0)
ld: warning: object file (/Users/runner/work/pulsar-client-node/pulsar-client-node/pkg/mac/build-pulsar/install/lib/libpulsarwithdeps.a(json_stream_parser.cc.o)) was built for newer macOS version (12.6) than being linked (11.0)
```

But the strange thing is that `mmacosx-version-min` is specified as 10.15 on the [cpp side](https://github.com/apache/pulsar-client-cpp/blob/main/pkg/mac/build-static-library.sh#L26).

Reply:
This warning is just because your platform that builds the Node.js client is 12.6. This warning should not be an issue.

Reply:
Maybe it's because of the config here? https://github.com/apache/pulsar-client-node/blob/801e5cea4a1cb4e218773b2cbb226a7518a25385/binding.gyp#L51